### PR TITLE
Support document symbol on ActiveSupport tests

### PR DESCRIPTION
**Pull request by st0012**
This feature allows users to navigate between AS tests with the `Go to Symbol` feature of VS Code.

It shares similar implementation with Code Lens around test case name extraction. And due to the complexity of the logic, I decided to extract it to a helper even though it's only shared between 2 components at the moment.

## Demo

document_symbol

### CODE REVIEW

File: lib/ruby_lsp/ruby_lsp_rails/document_symbol.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyLsp
+  module Rails
+    # ![Document Symbol demo](../../document_symbol.gif)
+    #
+    # The [document symbol](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_documentSymbol)
```
Comment (vinistock) [inline lib/ruby_lsp/ruby_lsp_rails/document_symbol.rb:8]: Let's not include the version and always link to the latest spec.
```suggestion
    # The [document symbol](https://microsoft.github.io/language-server-protocol/specification#textDocument_documentSymbol)
```

File: lib/ruby_lsp/ruby_lsp_rails/document_symbol.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      include ActiveSupportTestCaseHelper
+
+      sig do
+        params(
+          response_builder: ResponseBuilders::DocumentSymbol,
+          dispatcher: Prism::Dispatcher,
+        ).void
+      end
+      def initialize(response_builder, dispatcher)
+        @_response = T.let(nil, NilClass)
```
Comment (vinistock) [inline lib/ruby_lsp/ruby_lsp_rails/document_symbol.rb:22]: Do we need this variable?

**Review by vinistock — APPROVED** [review]
🚀

File: lib/ruby_lsp/ruby_lsp_rails/document_symbol.rb (hunk 2 above)
Comment (st0012) [reply lib/ruby_lsp/ruby_lsp_rails/document_symbol.rb]: Removed 👍

**Pull request merged**